model-builder/src/builder.rs

```diff
@@ -226,23 +242,24 @@ impl<'a> VehicleJourneyBuilder<'a> {
     /// ```
     /// # fn main() {
     /// let model = model_builder::ModelBuilder::default()
+    ///        .route("1", |r| {
+    ///            r.name = "bob".into();
+    ///        })
     ///        .vj("toto", |vj_builder| {
-    ///            vj_builder
-    ///                .route("1", |r| {
-    ///                    r.name = "bob".into();
-    ///                });
+    ///            vj_builder.route("1");
     ///        })
     ///        .build();
     /// # }
     /// ```
-    pub fn route<F>(mut self, name: &str, f: F) -> Self
-    where
-        F: Fn(&mut Route) + 'static,
-    {
-        self.route_modifier = Some(ObjectModifier {
-            id: name.to_owned(),
-            modifier: Box::new(f),
-        });
+    pub fn route(self, id: &str) -> Self {
+        {
+            let vj = &mut self
+                .model
+                .collections
+                .vehicle_journeys
+                .index_mut(self.vj_idx);
```

Comment:
you can use `.vehicle_journeys[self.vj_idx];`

Reply:
I don't think so, I need a mutable vj here.

Reply:
Yeah, my bad, `index_mut` on `CollectionWithId` is custom.